docs/frameworks/mastra.mdx

```diff
@@ -0,0 +1,318 @@
+---
+title: "Mastra"
+description: "Use Humanlayer with Mastra, the TypeScript AI Framework"
+icon: "solar-system"
+---
+
+## Overview
+
+[Mastra](https://mastra.ai) is a TypeScript framework for building AI applications. HumanLayer adds human oversight to your AI features.
+
+## Getting Started with Mastra
+
+### Installation
+
+The easiest way to get started with Mastra is to use `create-mastra`:
+
+```bash
+npx create-mastra@latest
+```
+
+This command will guide you through setting up a new Mastra project with all the necessary configurations.
+
+<Tip>
+  For more installation and project set up details, check Mastra's documentation
+  [here](https://mastra.ai/docs)
+</Tip>
+
+### Model Providers
+
+<Info>
+  While Google's Gemini models are used in the examples below, Mastra supports
+  all models available through the AI SDK:
+</Info>
+
+- OpenAI: `import { openai } from "@ai-sdk/openai"`
+- Anthropic: `import { anthropic } from "@ai-sdk/anthropic"`
+- Google: `import { google } from "@ai-sdk/google"`
+- And more...
+
+### Running Your Project
+
+Once your project is set up, you can run it using:
+
+```bash
+npm run dev
+```
+
+This will start your Mastra application, and you can access the playground at:
+
+```
+http://localhost:4111
+```
+
+## Tool Support
+
+Mastra provides flexible options for integrating tools into your AI agents.
+
+### Vercel AI SDK Tool Format
+
+<Info>
+  Mastra supports tools created using the Vercel AI SDK format. The HumanLayer
+  integration with Mastra uses this tool format.
+</Info>
+
+```typescript src/mastra/tools/vercelTools.ts
+import { tool } from "ai";
+import { z } from "zod";
+
+export const weatherInfo = tool({
+  description: "Fetches the current weather information for a given city",
+  parameters: z.object({
+    city: z.string().describe("The city to get weather for"),
+  }),
+  execute: async ({ city }) => {
+    // Replace with actual API call
+    const data = await fetch(`https://api.example.com/weather?city=${city}`);
+    return data.json();
+  },
+});
+```
+
+You can use Vercel tools alongside Mastra tools in your agents:
+
+```typescript src/mastra/agents/weatherAgent.ts
+import { Agent } from "@mastra/core/agent";
+import { openai } from "@ai-sdk/openai";
+import { weatherInfo } from "../tools/vercelTools";
+import * as mastraTools from "../tools/mastraTools";
+
+export const weatherAgent = new Agent({
+  name: "Weather Agent",
+  instructions:
+    "You are a helpful assistant that provides weather information.",
+  model: openai("gpt-4"),
+  tools: {
+    weatherInfo, // Vercel tool
+    ...mastraTools, // Mastra tools
+  },
+});
+```
+
+<Note>
+  Both tool formats will work seamlessly within your agent's workflow.
+</Note>
+
+## Basic Example
+
+Here's a simple example showing how to use HumanLayer with Mastra:
+
```

Comment:
So far, this doesn't feel like a step-by-step guide - for the best chance of user success (with both HL *and* mastra) I'd love to have a little more direction like "hey add X to file Y". I don't know where to put this file, I just have the example.

Reply:
also, the steps above, it seems now like those are just demonstrative, and I'm not actually expected to put those files anywhere. This was confusing, esp because they have file names in the code block